src/dash/models/DashManifestModel.js

```diff
@@ -773,6 +787,56 @@ function DashManifestModel() {
         return utcTimingEntries;
     }
 
+    function getBaseURLsFromElement(node) {
+        let baseUrls = [];
+        let entries = node.BaseURL_asArray || [node.baseUri] || [];
+
+        entries.some(entry => {
+            if (entry) {
+                const baseUrl = new BaseURL();
+                const text = entry.__text || entry;
+
+                baseUrl.url = text;
+
+                // serviceLocation is optional, but we need it in order
+                // to blacklist correctly. if it's not available, use
+                // anything unique since there's no relationship to any
+                // other BaseURL and, in theory, the url should be
+                // unique so use this instead.
+                if (entry.hasOwnProperty('serviceLocation') &&
+                        entry.serviceLocation.length) {
+                    baseUrl.serviceLocation = entry.serviceLocation;
+                } else {
+                    baseUrl.serviceLocation = text;
+                }
+
+                if (entry.hasOwnProperty('dvb:priority')) {
+                    baseUrl.dvb_priority = entry['dvb:priority'];
+                }
+
+                if (entry.hasOwnProperty('dvb:weight')) {
+                    baseUrl.dvb_weight = entry['dvb:weight'];
+                }
+
+                /* NOTE: byteRange, availabilityTimeOffset,
+                 * availabilityTimeComplete currently unused
+                 */
+
+                baseUrls.push(baseUrl);
+
+                if (urlUtils.isRelative(text)) {
+                    // it doesn't really make sense to have relative and
+                    // absolute URLs at the same level, or multiple
+                    // relative URLs at the same level, so assume we are
+                    // done from this level of the MPD
```

Comment:
I think it could potentially make sense to have relative and absolute URLs at the same level - you could have the first URL as relative (i.e. continue loading requests from the same domain as the MPD), then fallback to mirrors on other domains. This is more thinking out loud than something I'd want to do myself though.